opentelemetry-api/src/opentelemetry/util/__init__.py

```diff
@@ -53,15 +52,9 @@ def _load_provider(
         raise
 
 
-# Pattern for matching up until the first '/' after the 'https://' part.
-_URL_PATTERN = r"(https?|ftp)://.*?/"
-
-
-def disable_tracing_path(url: str, excluded_paths: Sequence[str]) -> bool:
-    if excluded_paths:
-        # Match only the part after the first '/' that is not in _URL_PATTERN
-        regex = "{}({})".format(_URL_PATTERN, "|".join(excluded_paths))
-        if re.match(regex, url):
+def disable_tracing_path(url_path: str, excluded_paths: Sequence[str]) -> bool:
```

Comment:
If I understand this correctly, I think we may have an issue here. If we add `aaa` to the excluded paths, then any host that has `aaa` in the beginning of their paths will be excluded. But if we want to do exclusion for a series of paths that begin with `aaa` for a single host, then we need to list them all in the `*_EXCLUDED_HOSTS` environment variable. If we want to have both things, per-host exclusion and multiple exclusion for any path that begins with a particular string, I think we should have only the `*_EXCLUDED_HOSTS` environment variable but make it behave as the `*_EXCLUDED_PATHS`, which can exclude many paths that begin with a certain string.

Reply:
How can we achieve the same granularity of checking with the proposed solution (ie., still be able to exclude exact hosts)? What if I want to do exclude http://example.com/test but don't want to exclude http://example.com/stage?

Reply:
Then you add `http://example.com/test` to `OPENTELEMETRY_PYTHON_EXCLUDED_HOSTS`.

Reply:
But then what if we want to exclude `http://example.com/test/foo` but not `http://example.com/test/bar`? We will still need two levels of control, one for exact matching and another with a "starts with" behaviour.

Reply:
Yes, but in that case I think we would need 2 environment variables, one for controlling the "starts with" and another one for "matches exactly", right?

Reply:
Yes exactly. However I don't know if we should have a "starts with" that accepts everything. We should inform the user that the "starts with" will match something conceptual (like a path), otherwise they will be entering any string that they want (matching "http:/" doesn't make any sense).

Reply:
I'm not totally persuaded that matching something like `"http:/"` is much of a problem. I understand that it does not make much logical sense, but it would still work for our purposes, right?
If that is still an obstacle for you, how about ignoring anything in the environment variables that does not end with `/` or does not have at least one element after the protocol declaration?

For example, ignore `http:/`, `htt`, `http://`, `http://someth`. Accept `http://something/`. That will mean, of course, that the user will need to add in the environment variable each protocol that is to be excluded along with the host (for example `https://something/` would need to be included too), but I guess to solve that problem we would need to use regexes :sweat_smile:

Reply:
Alright, I added the regex logic back in for paths.